soc/arm/nxp_imx/rt/soc_rt10xx.c

```diff
@@ -14,75 +14,25 @@
 #include <arch/cpu.h>
 #include <arch/arm/aarch32/cortex_m/cmsis.h>
 #include <fsl_flexspi_nor_boot.h>
+#include <clock_config.h>
```

Comment:
This assumes that there is `clock_config.h` already provided. There are several issues with it:
- at the point when this commit is introduced, there are no board specific `clock_config.h` files (they are introduced in subsequent commits), so this change breaks bisectability of the codebase,
- this change enforces new platforms (boards) to always come with generated clock configuration, which is additional level besides device-tree to configure; not sure that is desirable,
- including board specific file from SoC code seems a bit hacky; I think that having a default configuration for SoC is much easier for introducing new boards in the future, while providing an option for board/application specific fine-tuned clock configuration.

Reply:
@mmahadevan108, will need to arrange the commits to ensure that the clock_config.h exists before introducing it in this file for the bisectability.

Reply:
> * this change enforces new platforms (boards) to always come with generated clock configuration, which is additional level besides device-tree to configure; not sure that is desirable,

Currently, the clock configuration code are the lines being deleted from this file. The way it has previously been defined in this file was very error prone.

Reply:
I have rearranged the commits to address the issue with `git bisect`

Reply:
What was that error-prone previously and has been improved as part of this PR? This is not clear to me.

Having `#include <clock_config.h>` here is a soft requirement to use code generation from MCUxpresso config tool, because that tool actually creates `clock_config.h` file together with implementation of `clock_init()` function (which is part of generated `clock_config.c`. Even though using code generation is not a strict requirement, developer needs to mimic the same behavior (implement `clock_init()` function and provide `clock_config.h` header), which seems weird.